command-line/lib/Command/Address/Reward.hs

```diff
@@ -68,15 +70,21 @@ mod liftCmd = command "stake" $
             ])
   where
     parser = Cmd
-        <$> networkTagOpt Shelley
+        <$> credentialOpt
+        <*> networkTagOpt Shelley
 
 run :: Cmd -> IO ()
-run Cmd{networkTag} = do
-    xpub <- hGetXPub stdin
+run Cmd{networkTag,credential} = do
     case (mkNetworkDiscriminant . fromIntegral . unNetworkTag) networkTag of
         Left e -> die (fmt $ build e)
         Right discriminant -> do
-            let addr = Shelley.stakeAddress discriminant (Shelley.liftXPub xpub)
+            addr <- case credential of
+                    CredentialFromKey -> do
+                        xpub <- hGetXPub stdin
+                        pure $ Shelley.stakeAddress discriminant (Shelley.FromKey $ Shelley.liftXPub xpub)
+                    CredentialFromScript -> do
+                        scriptHash <- hGetScriptHash stdin
+                        pure $ Shelley.stakeAddress discriminant (Shelley.FromScript scriptHash)
```

Comment:
For consistency, wouldn't it be better that we accept a _script_ has an input here, and simply store it as a hash ?